mvc/overview.rst

```diff
@@ -1,8 +1,210 @@
+
+.. code-block:: c#
+  :emphasize-lines: 1,4
+
+  [Route("api/[controller]")]
+  public class ProductsController : Controller
+  {
+    [HttpGet("{id}")]
+    public IActionResult GetProduct(int id)
+    {
+      ...
+    }
+  }
+
+Model binding
+-------------
+
+MVC :doc:`model binding </models/model-binding>` converts request data (form values, route data, query string parameters, HTTP headers) into objects that the controller can handle. As a result, your controller logic doesn't have to do the work of figuring out the incoming request data; it simply has the data as parameters to its action methods.
+
+.. code-block:: C#
+
+  public async Task<IActionResult> Login(LoginViewModel model, string returnUrl = null) { ... }
+
+Request validation
+------------------
+
+MVC provides a powerful :doc:`validation <models/validation>` system for validating request data. Simply add validation logic to your model types using data annotation attributes:
+
+.. code-block:: c#
+  :emphasize-lines: 3-4,7-8
+
+  public class LoginViewModel
+  {
+      [Required]
+      [EmailAddress]
+      public string Email { get; set; }
+
+      [Required]
+      [DataType(DataType.Password)]
+      public string Password { get; set; }
+
+      [Display(Name = "Remember me?")]
+      public bool RememberMe { get; set; }
+  }
+
+MVC will handle validating request data both on the client and on the server. Validation logic specified on model types is added to the rendered views as unobtrusive annotations and is enforced in the browser via `jQuery Validation <http://jqueryvalidation.org/>`__.
+
+Razor View Engine
+-----------------
+
+MVC uses the Razor view engine for rendering views. Razor is a compact, expressive and fluid template markup language for defining views using embedded .NET code. Razor is used to dynamically generate web content on the server. Unlike most template syntaxes, you do not need to interrupt your coding to explicitly denote server blocks within your HTML. The parser is smart enough to infer this from your code. This make creating views in Razor clean, fast and fun!
+
+.. code-block:: html
+
+  <ul>
+    @for (int i = 0; i < 5; i++) {
+      <li>List item @i</li>
+    }
+  </ul>
+
+Using the MVC Razor view engine you can define layouts, partial views and replaceable sections.
+
+Strongly typed views
+--------------------
+
+Razor views in MVC can be strongly typed based on your model. A strongly-typed view explicitly specifies the model type for the view. You can then leverage the compiler and IntelliSense to ensure that references to property names and types in your views are correct:
+
+.. code-block:: html
+
+  @model IEnumerable<MvcMusicStore.Models.Album>
+  <ul> 
+      @foreach (Album p in Model) 
+      { 
+          <li>@p.Title</li>
+      }
+  </ul>
+
+HTML Helpers
```

Comment:
@Rick-Anderson Tag helpers are new and closer to HTML syntax but are not intended as a complete replacement for HTML helpers -- more a replacement for or at least alternate to _some_ HTML helpers. Many HTML helpers (`ValueFor()`, `EditorFor()` and more) have no tag helper equivalents because either they generate a single `string` or they don't generate predictable HTML content.